docs/src/main/sphinx/develop/supporting-merge.rst

```diff
@@ -0,0 +1,390 @@
+  ``ChangeOnlyUpdatedColumnsMergeProcessor``.
+* ``DELETE_ROW_AND_INSERT_ROW``, intended for connectors like ``trino-hive``
+  and ``trino-iceberg`` that represent a row change as a row deletion paired
+  with a row insertion.  The corresponding merge processor class is
+  ``DeleteAndInsertMergeProcessor``.
+
+Overview of ``MERGE`` processing
+--------------------------------
+
+A ``MERGE`` statement is processed by creating a ``RIGHT JOIN`` between the
+target table and the source, on the ``MERGE`` criteria.  The source may be
+a table or an arbitrary query.  For each row in the source table or query,
+``MERGE`` produces a ``ROW`` object containing:
+
+* the column values from the ``UPDATE`` or ``INSERT`` cases, or ``NULL``
+  for the ``DELETE`` cases
+* an integer providing the case number of the case that matched the source
+  row, or -1 meaning that no case matched the source row
+* an integer that identifies whether the merge case operation is ``UPDATE``,
+  ``DELETE`` or ``INSERT``.
+
+The example above is executed as if it were written as::
+
+   SELECT
+    FROM (SELECT *, true AS present FROM target_table) t
+    RIGHT JOIN source_table s ON s.customer = t.customer
+    CASE
+      WHEN present AND s.address = 'Berkeley' THEN
+          // Null values for delete; case #0; operation DELETE=2
+          row(null, null, null, 0, 2)
+      WHEN present AND s.customer = 'Joe Shmoe' THEN
+          // Update column values; case #1; operation UPDATE=3
+          row(t.customer, t.purchases + 100.0, t.address, 1, 3)
+      WHEN present THEN
+          // Update column values; case #2; operation UPDATE=3
+          row(t.customer, s.purchases + t.purchases, s.address, 2, 3)
+      WHEN (present IS NULL) THEN
+          // Insert column values; case #3; operation INSERT=1
+          row(s.customer, s.purchases, s.address, 3, 1)
+      ELSE
+          // Null values for no case matched; case #-1; operation=-1
+          row(null, null, null, -1, -1)
+    END;
+
+The Trino engine executes the ``RIGHT JOIN`` and ``CASE`` expression,
+creating a sequence of pages to be routed to the node that runs the
+``ConnectorMergeSink``.``storeMergedRows(...)`` method.  To ensure that all
+inserted rows for a given partition end up on a single node, a
+redistribution hash on the partition key/bucket column(s) is applied
+to the page partition key(s).  As a result of the hash, all rows for a
+specific partition/bucket will hash together, whether they were
+``MATCHED`` rows or ``NOT MATCHED`` rows.
+
+Like ``DELETE`` and ``UPDATE``, ``MERGE`` target table rows are identified by
+a connector-specific ``rowId`` column handle, returned by
+``ConnectorMetadata.getMergeRowIdColumnHandle(...)``.  For ``MERGE``,
+that ``rowId`` column contains whatever the connector needs in order
+to process all the possible ``MERGE`` cases for that row.  For example,
+in the case of the Hive connector, the merge handle contains the
+``InsertTableHandle`` for the target table.
```

Comment:
No, it's correct: the `MergeTableHandle` contains the `InsertTableHandle`.  It must do so, because the only entity passed back to the `ConnectorMergeSink` is the `MergeTableHandle`.

Reply:
You're right, it was a jumble.  Reduced to just:

```
Like ``DELETE`` and ``UPDATE``, ``MERGE`` target table rows are identified by
a connector-specific ``rowId`` column handle, returned by
``ConnectorMetadata.getMergeRowIdColumnHandle(...)``.
```